src/app/ui/editor/moving_pixels_state.cpp

```diff
@@ -597,6 +597,29 @@ void MovingPixelsState::onBeforeLayerVisibilityChange(Editor* editor,
   }
 }
 
+void MovingPixelsState::onBeforeLayerEditableChange(Editor* editor,
+                                                    doc::Layer* layer,
+                                                    bool newState)
+{
```

Comment:
Now I think that since `onBeforeLayerEditableChange` and `onBeforeLayerVisibilityChange` do the same, you could create a private method that would be called by both. With this I'm saying that in my previous review what I didn't like was changing the interface of DocObserver. But I do like to avoid code duplication when possible.